backend/tests/controllers/reports/test_reports_test_executions.py

```diff
@@ -0,0 +1,197 @@
+    environment = generator.gen_environment()
+    generator.gen_test_execution(
+        artefact_build_1, environment, created_at=datetime.now()
+    )
+    test_execution = generator.gen_test_execution(
+        artefact_build_2, environment, created_at=datetime.now()
+    )
+    generator.gen_artefact_build_environment_review(artefact_build_1, environment)
+    generator.gen_artefact_build_environment_review(artefact_build_2, environment)
+
+    response = test_client.get(
+        "/v1/reports/test-executions",
+        params={
+            "start_date": (datetime.now() - timedelta(days=1)).isoformat(),
+            "end_date": datetime.now().isoformat(),
+        },
+    )
+
+    table = _read_csv_response(response)
+
+    assert len(table) == 2
+    assert table[0] == [str(c) for c in TEST_EXECUTIONS_REPORT_COLUMNS]
+    assert table[1] == _expected_report_row(test_execution, db_session=db_session)
+
+
+def test_get_testexecutions_report_with_test_events(
+    test_client: TestClient, generator: DataGenerator, db_session: Session
+):
+    artefact = generator.gen_artefact("beta")
+    artefact_build = generator.gen_artefact_build(artefact)
+    environment = generator.gen_environment()
+    test_execution = generator.gen_test_execution(
+        artefact_build, environment, created_at=datetime.now()
+    )
+    generator.gen_artefact_build_environment_review(artefact_build, environment)
+    generator.gen_test_event(test_execution, "provision_fail")
+
+    response = test_client.get(
+        "/v1/reports/test-executions",
+        params={
+            "start_date": (datetime.now() - timedelta(days=1)).isoformat(),
+            "include_test_events": True,
+        },
+    )
+
+    table = _read_csv_response(response)
+
+    assert len(table) == 2
+    assert table[0] == [
+        str(column)
+        for column in [
+            *TEST_EXECUTIONS_REPORT_COLUMNS,
+            "TestEvent.event_name",
+            "TestEvent.timestamp",
+            "TestEvent.detail",
+        ]
+    ]
+    assert table[1] == _expected_report_row(
+        test_execution, db_session=db_session, include_test_events=True
+    )
+
+
+def _read_csv_response(response: Response) -> list:
+    content = response.content.decode()
+    csv_reader = csv.reader(StringIO(content))
+    return list(csv_reader)
+
+
+def _expected_report_row(
+    test_execution: TestExecution,
+    db_session: Session,
+    include_test_events: bool = False,
+) -> list:
+    environment = test_execution.environment
+    artefact = test_execution.artefact_build.artefact
+    family = artefact.stage.family
+    environment_review = (
+        db_session.query(ArtefactBuildEnvironmentReview)
+        .filter(
+            ArtefactBuildEnvironmentReview.artefact_build_id
+            == test_execution.artefact_build_id,
+            ArtefactBuildEnvironmentReview.environment_id
+            == test_execution.environment_id,
+        )
+        .one()
+    )
```

Comment:
I replaced the `session.query` with `session.execute` and a `select` statement. I decided to do it this way because for one ArtefactBuild there might be many ArtefactBuildEnvironmentReview (in case of multiple environments). Although in our case we have just one, I decided to avoid filtering in Python, and to make a separate query in the database.